### ISSUE TITLE
One of lawsets have two laws 3

### ISSUE THREAD

**Issue by thebadman4662** [space-wizards/space-station-14#33183]
**Screenshots**
![law33](https://github.com/user-attachments/assets/26c171f9-161c-4489-890c-81225380d1fd)

**Comment by Thatonestomf** [comment]
ah that is intended, this one is NT default i think, its kinda like crewsimov, but with rank and job in mind

**Comment by Thatonestomf** [comment]
oh wait, just saw the numberin.... uh... hmmm
well that should be an easy enough fix, imma do it myself

**Comment by thebadman4662** [comment]
> ah that is intended, this one is NT default i think, its kinda like crewsimov, but with rank and job in mind

![obraz](https://github.com/user-attachments/assets/e19d8c4d-b550-4cb0-a972-1433d7cc2aea)
Nothing in rules written about non-ion laws with same number but different order... is AI free to choose and can break one over other at will?

**Comment by Thatonestomf** [comment]
nope, im just BLIND, and didnt notice the numbers on the side being wrong fixin it now

**Comment by lzk228** [comment]
> nope, im just BLIND, and didnt notice the numbers on the side being wrong fixin it now

so how is fixing? :godo:

**Comment by Thatonestomf** [comment]
> so how is fixing? :godo:

Awful I gave up day one, I had no clue what I was doing

**Comment by LordCarve** [comment]
Gone over all commit history to `silicon-law.yaml` file since the beginning of time:
No lawsets ever had laws with duplicate order or `LawIdentifierOverride` set.
***IF*** there is an issue, it is not a prototype issue, but logic issue.

So down the logic rabbit hole I went...

That value (law number) can only be `Order` or `LawIdentifierOverride`.
There is only three ways to modify these values:
- Manage Laws UI _(Admin only)_ - **`Order` can be set to duplicate values.** i.e. Law 3 twice, as on screenshot ❌; `LawIdentifierOverride` is only a scrambled/unscrambled toggle (no plaintext/numeric overrides) ✔.
- Emag - Under normal circumstances emagging will cause no duplicate `Order` values (usable once per target; adds rule 0 and non-conflicting one) ✔; If Law 0 already exists, will create a law with duplicate `Order` 0 ❌.
- Ion storm - **It uses `LawIdentifierOverride` very liberally** _(for final reordering it should be using `Order` instead)_, which is bad on its own, but the current logic as-is cannot result in duplicate values on its own ✔ _(in fact, it will avoid duplicates and fix duplicates if existing before event triggered)._

So in short: **This can be replicated, but only with Admin intervention.**

Steps to replicate:
1. Find a silicon (for exact laws on screenshot, use a borg with `nt-default` lawset)
2. Admin verb -> Manage Laws
3. Click the `-1` on one of the later laws, then `Save changes` (Law 4 if going for exact replication)
4. **Silicon now has two laws with same order. (Bug replicated)**

Next steps:
As already pointed out in this thread, Rules do not anticipate lawset ever having multiple laws with the exact same priority (Silicon Rules specifically). This is therefore confusing for players (as evidenced by comments here) with severity further compounded by the fact that rules can be enforced OOC (punishments for violating rules). This creates problematic and potentially ahelp situations (which law takes precedence? Can I choose? Am I risking a ban if I interpret the order of the rules incorrectly?)

I believe the implied intent in Silicon Rules (intentional omission of anything pertaining to same-priority laws - it is not vague or permissive - ***it is non-existent***) is that as a game-enforced rule you will never have two or more laws with the same priority.

In light of the above, and in the same spirit as #20064 _(although not really a mechanic but more of a core function)_, the correct way to address this problem is to:
- At minimum: make it impossible for Admins/Emags to create priority-conflicing laws (multiple laws with same priority);
- Recommended: implement logic guards that ensure all laws within a lawset **MUST ALWAYS** use different priorities (code cleanliness & future feature proofing).

------

I plan to work on this.
